src/horus_api.h

```diff
@@ -35,20 +35,75 @@
 #include <stdint.h>
 #include "modem_stats.h"
       
-#define HORUS_MODE_BINARY            0
-#define HORUS_MODE_RTTY              1
+/* Horus API Modes */
+#define HORUS_MODE_BINARY            0  // Legacy binary mode
+#define HORUS_MODE_BINARY_256BIT     1  // New 256-bit LDPC-encoded mode
+#define HORUS_MODE_BINARY_128BIT     2  // New 128-bit LDPC-encoded mode
+#define HORUS_MODE_RTTY              99 // RTTY Decoding
+
+
+// Settings for Legacy Horus Binary Mode (Golay Encoding)
+#define HORUS_BINARY_NUM_BITS                   360   /* fixed number of coded bits */
+#define HORUS_BINARY_NUM_PAYLOAD_BYTES          22    /* fixed number of bytes in binary payload     */
+#define HORUS_BINARY_DEFAULT_BAUD               100   /* Default baud rate of 100 baud */
+#define HORUS_BINARY_DEFAULT_TONE_SPACING       270   /* Default tone spacing of 270 Hz */
+
+// Settings for Horus Binary 256-bit mode (LDPC Encoding, r=1/3)
+#define HORUS_BINARY_256BIT_NUM_BITS            768
+#define HORUS_BINARY_256BIT_NUM_PAYLOAD_BYTES   32
+#define HORUS_BINARY_256BIT_DEFAULT_BAUD        100   /* Default baud rate of 100 baud */
+#define HORUS_BINARY_256BIT_DEFAULT_TONE_SPACING 270   /* Default tone spacing of 270 Hz */
+
+// Settings for Horus Binary 128-bit mode (LDPC Encoding, r=1/3)
+#define HORUS_BINARY_128BIT_NUM_BITS            384
```

Comment:
So is this just the FEC codeword, excluding the UW?

Reply:
Good point, not sure if this should include the UW bits or not. Will investigate furthere.

Reply:
Aha, further down the buffer is set to double the number of expected bits: https://github.com/drowe67/codec2/blob/044daa39249c2c94470f0f1e1ec973409f0de71a/src/horus_api.c#L177

This ensures there's enough space for a UW and a packet.